src/test/java/net/wasdev/wlp/common/plugins/util/DevUtilTest.java

```diff
@@ -76,12 +85,150 @@ public void testCleanupServerEnvBak() throws Exception {
         assertTrue(serverEnvBak.exists());
 
         util.cleanUpServerEnv();
-        
+
         // verify the backup env file was restored as server.env
         assertTrue(serverEnv.exists());
         String serverEnvContents = new String(Files.readAllBytes(serverEnv.toPath()));
         assertEquals("backup", serverEnvContents);
         assertFalse(serverEnvBak.exists());
     }
 
+    @Test
+    public void testReadFileToString() throws Exception {
+        DevUtil util = new DevTestUtil(null, null, null, null, null, false);
+
+        File serverDirectory = Files.createTempDirectory("serverDirectory").toFile();
+        File tempFile = new File(serverDirectory, "temp.txt");
+        Files.write(tempFile.toPath(), "temp".getBytes());
+        String fileString = util.readFile(tempFile);
+
+        assertTrue(fileString.equals("temp"));
+
+        if (tempFile.exists()) {
+            tempFile.delete();
+        }
+    }
+
+    @Test
+    public void testCopyConfigFolder() throws Exception {
+    }
+
+    @Test
+    public void testCopyFile() throws Exception {
+        DevUtil util = new DevTestUtil(null, null, null, null, null, false);
+
+        File srcDir = Files.createTempDirectory("src").toFile();
+        File targetDir = Files.createTempDirectory("target").toFile();
+        File configFile = new File(srcDir, "config.xml");
+        Files.write(configFile.toPath(), "temp".getBytes());
+
+        util.copyFile(configFile, srcDir, targetDir, "server.xml");
+
+        File targetFile = new File(targetDir, "server.xml");
+        assertTrue(targetFile.exists());
+
+        // clean up
+        if (srcDir.exists()) {
```

Comment:
If the test fails before this point, the cleanup won't be run.  It's better to cleanup using in a `finally` block or in the tearDown() method.